mypy/subtypes.py

```diff
@@ -150,19 +186,17 @@ def visit_instance(self, left: Instance) -> bool:
             return True
         right = self.right
         if isinstance(right, TupleType) and right.fallback.type.is_enum:
-            return is_subtype(left, right.fallback)
+            return self._is_subtype(left, right.fallback)
         if isinstance(right, Instance):
-            if TypeState.is_cached_subtype_check(left, right):
+            if self._lookup_cache(left, right):
                 return True
             # NOTE: left.type.mro may be None in quick mode if there
             # was an error somewhere.
-            if left.type.mro is not None:
+            if not self.ignore_promotions and left.type.mro is not None:
                 for base in left.type.mro:
                     # TODO: Also pass recursively ignore_declared_variance
```

Comment:
I think this TODO can be removed since `_is_subtype` does this automatically.

Reply:
Done.